src/engine/renderer/Material.cpp

```diff
@@ -2051,6 +2052,80 @@ void MaterialSystem::RenderMaterial( Material& material, const uint32_t viewID )
 
 	atomicCommandCountersBuffer.BindBuffer( GL_PARAMETER_BUFFER_ARB );
 
+	if ( r_showGlobalMaterials.Get() && material.sort != 0
+		&& ( material.shaderBinder == BindShaderLightMapping || material.shaderBinder == BindShaderGeneric3D ) ) {
+		vec3_t color;
+		/* Some simple random modifiers to make the colors more contrasting
+		Maybe we can use some better way of assigning colors here? */
+		static vec3_t colors[6] = { { 0.75, 0.25, 0.25 }, { 0.75, 0.75, 0.25 }, { 0.25, 0.75, 0.25 }, { 0.25, 0.75, 0.75 },
+			{ 0.25, 0.25, 0.75 }, { 0.75, 0.25, 0.75 } };
+
+		switch ( r_showGlobalMaterials.Get() ) {
+			case Util::ordinal( MaterialDebugMode::DEPTH ):
+			{
+				// Even though this is for depth materials, we don't actually draw anything on depth pass
+				if ( material.sort != 1 ) {
+					return;
+				}
+
+				const float id = ( float ) material.id / ( materialPacks[0].materials.size() + 2 ) + 1;
+
+				color[0] = fmod( id, 1 / 3.0 ) * colors[int( material.id * 6.0
+					/ materialPacks[0].materials.size() )][0];
+				color[1] = fmod( std::max( 0.0, id - 1 / 3.0 ), 1 / 3.0 ) * colors[int( material.id * 6.0
```

Comment:
What are these color calculations trying to do? This part does (id - 1/3) modulo 1/3 which doesn't make sense as that's just id modulo 1/3.

Reply:
The idea was to assign a different rgb colour to each material. The calculation is definitely wrong though looking at it again, gonna fix it later.

Reply:
Fixed.